include/exadg/operators/operator_base.cpp

```diff
@@ -380,19 +385,31 @@ OperatorBase<dim, Number, n_components>::evaluate_add(VectorType &       dst,
   }
   else
   {
+    AssertThrow(data.dof_index_inhomogeneous != dealii::numbers::invalid_unsigned_int,
+                dealii::ExcMessage("dof_index_inhomogeneous is uninitialized."));
+
+    // Start with a zero destination vector.
+    VectorType tmp;
+    tmp.reinit(dst, false);
+
+    // Perform matrix-vector product using src_inhom (which contains inhomogeneous Dirichlet values)
+    // as src-vector to obtain the hom+inhom action of the operator.
     if(evaluate_face_integrals())
     {
-      matrix_free->loop(&This::cell_loop,
+      matrix_free->loop(&This::cell_loop_full_operator,
                         &This::face_loop_empty,
                         &This::boundary_face_loop_inhom_operator,
                         this,
-                        dst,
+                        tmp,
                         src);
```

Comment:
Question: If you run `MatrixFree::loop` or `MatrixFree::cell_loop`, the default behavior is to add into the `dst` vector. This means that we should be able to avoid the temporary vector here and thus save a bit of memory capacity and memory transfer (the extra copy does 4 vector accesses). We do the same in the DG case above.

Reply:
I guess this code is some leftover, because we had - at least temporarily - also other functionality in this function.

Reply:
see https://github.com/exadg/exadg/pull/506/commits/c45b4bbd3acac43cbc8f0059d07505f60bd629fa